#Generation of Counts Matrices - This script compiles the counts from the HTseq counts files generated by the alignment.

#Prior to this script I chose only cells which were single and live on the chip (not 1r)
#Only sequencing from MiSeq

#All of the cells were placed into the folder (AllCounts) which was called when compiling the counts.

#Filtering for high quality cells was achieved by filtering for 20,000 reads mapping to transcriptome and 500 genes detected.


rm(list=ls())
library(ggplot2)
library(edgeR)


# All specific pops ---------------------------------------------------------------------

#Reading in all HTseq counts files.
setwd("/Volumes/guacamole/Software/Final Scripts for Paper _ 01152015/AllCounts/")
fileList=list.files(pattern=".counts.txt$")

#Combine the counts from each of the individual files to make one large counts matrix and save the counts matrix to
#a file.
allcounts<-read.table(fileList[1],header=F)
colnames(allcounts)[2]<-strsplit(fileList[1],split="[.]")[[1]][1]

for (i in 2:length(fileList)){
  temp = read.table(fileList[i],header=F)
  allcounts=cbind(allcounts,temp[,2])
  colnames(allcounts)[i+1]<-strsplit(fileList[i],split="[.]")[[1]][1]
}

allcounts<-allcounts[-c((length(allcounts[,1])-4):length(allcounts[,1])),] #removes HTseq QC values from bottom of counts matrix
allcounts_fin<-allcounts[,2:length(allcounts[1,])]
rownames(allcounts_fin)<-allcounts[,1]


#Read in all counts that have ERCC spike ins
setwd("/Volumes/guacamole/Software/Final Scripts for Paper _ 01152015/AllCounts/ERCC")
fileList=list.files(pattern=".counts.txt$")

#Combine the counts from each of the individual files to make one large counts matrix and save the counts matrix to
#a file.
allcounts_ERCC<-read.table(fileList[1],header=F)
colnames(allcounts_ERCC)[2]<-strsplit(fileList[1],split="[.]")[[1]][1]

for (i in 2:length(fileList)){
  temp = read.table(fileList[i],header=F)
  allcounts_ERCC=cbind(allcounts_ERCC,temp[,2])
  colnames(allcounts_ERCC)[i+1]<-strsplit(fileList[i],split="[.]")[[1]][1]
}

allcounts_ERCC<-allcounts_ERCC[-c((length(allcounts_ERCC[,1])-4):length(allcounts_ERCC[,1])),] #removes HTseq QC values from bottom of counts matrix
allcounts_ERCC_fin<-allcounts_ERCC[,2:length(allcounts_ERCC[1,])]
rownames(allcounts_ERCC_fin)<-allcounts_ERCC[,1]

#Removing ERCC spike ins from the cells which were mapped to the ERCC transcriptome, so that the matrices can be combined.
allcounts_noERCC_fin<-allcounts_ERCC_fin[!grepl("ERCC-",rownames(allcounts_ERCC_fin)),]

#Combines cells with ERCC spike ins and those that don't
allcounts_allcells<-cbind(allcounts_fin,allcounts_noERCC_fin)

#Wrote the counts to a file which I now read and continue the filtering process
setwd("/Volumes/guacamole/Software/Final Scripts for Paper _ 01152015/Files")
write.table(allcounts_allcells,"All_spec_pops_counts_09222015.txt")

#The following code changes the names to match the naming approach which I used to name the cells in previous analyses.
str_vec<-vector()
for(i in 1:length(colnames(allcounts_allcells))){
  str<-""
  if(strsplit(colnames(allcounts_allcells)[i],split=c("[_]"))[[1]][1]=="PGE120"){
    filename<-strsplit(colnames(allcounts_allcells)[i],split=c("[_]"))
    num<-strsplit(filename[[1]][2],split=c("[-]"))[[1]][2]
    qual<-strsplit(filename[[1]][2],split=c("[-]"))[[1]][3]
    str<-paste(str,"aNSC_C_",num,"_",qual,sep="")
    str_vec<-c(str_vec,str)
  }else if(strsplit(colnames(allcounts_allcells)[i],split=c("[_]"))[[1]][1]=="PGE210"){
    filename<-strsplit(colnames(allcounts_allcells)[i],split=c("[_]"))
    num<-filename[[1]][3]
    qual<-filename[[1]][4]
    str<-paste(str,"aNSC_D_",num,"_",qual,sep="")
    str_vec<-c(str_vec,str)
  }else if(strsplit(colnames(allcounts_allcells)[i],split=c("[_]"))[[1]][1]=="PG210"){
    filename<-strsplit(colnames(allcounts_allcells)[i],split=c("[_]"))
    num<-filename[[1]][3]
    qual<-filename[[1]][4]
    str<-paste(str,"qNSC_C_",num,"_",qual,sep="")
    str_vec<-c(str_vec,str)
  }else if(strsplit(colnames(allcounts_allcells)[i],split=c("[-]"))[[1]][1]=="Na"){
    filename<-strsplit(colnames(allcounts_allcells)[i],split=c("[-]"))
    num<-gsub("C","",strsplit(filename[[1]][2],split=c("[_]"))[[1]][1])
    str<-paste(str,"aNSC_A_",num,sep="")
    str<-paste(str,"_1g",sep="")
    str_vec<-c(str_vec,str)
  }else if(strsplit(colnames(allcounts_allcells)[i],split=c("[-]"))[[1]][1]=="Nq"){
    filename<-strsplit(colnames(allcounts_allcells)[i],split=c("[-]"))
    num<-gsub("C","",strsplit(filename[[1]][2],split=c("[_]"))[[1]][1])
    str<-paste(str,"qNSC_A_",num,sep="")
    str<-paste(str,"_1g",sep="")
    str_vec<-c(str_vec,str)
  }else if(strsplit(colnames(allcounts_allcells)[i],split=c("[-]"))[[1]][1]=="PGE"){
    filename<-strsplit(colnames(allcounts_allcells)[i],split=c("[-]"))
    num<-gsub("C","",strsplit(filename[[1]][2],split=c("[_]"))[[1]][1])
    str<-paste(str,"aNSC_B_",num,sep="")
    str<-paste(str,"_1g",sep="")
    str_vec<-c(str_vec,str)
  }else if(strsplit(colnames(allcounts_allcells)[i],split=c("[-]"))[[1]][1]=="PG"){
    filename<-strsplit(colnames(allcounts_allcells)[i],split=c("[-]"))
    num<-gsub("C","",strsplit(filename[[1]][2],split=c("[_]"))[[1]][1])
    str<-paste(str,"qNSC_B_",num,sep="")
    str<-paste(str,"_1g",sep="")
    str_vec<-c(str_vec,str)
  }else if(strsplit(colnames(allcounts_allcells)[i],split=c("[-]"))[[1]][1]=="G"){
    filename<-strsplit(colnames(allcounts_allcells)[i],split=c("[-]"))
    num<-gsub("C","",strsplit(filename[[1]][2],split=c("[_]"))[[1]][1])
    str<-paste(str,"Ast_",num,sep="")
    str<-paste(str,"_1g",sep="")
    str_vec<-c(str_vec,str)
  }else if(strsplit(colnames(allcounts_allcells)[i],split=c("[-]"))[[1]][1]=="E"){
    filename<-strsplit(colnames(allcounts_allcells)[i],split=c("[-]"))
    num<-gsub("C","",strsplit(filename[[1]][2],split=c("[_]"))[[1]][1])
    str<-paste(str,"NPC_",num,sep="")
    str<-paste(str,"_1g",sep="")
    str_vec<-c(str_vec,str)
  }else if(strsplit(colnames(allcounts_allcells)[i],split=c("[-]"))[[1]][1]=="N"){
    filename<-strsplit(colnames(allcounts_allcells)[i],split=c("[-]"))
    num<-gsub("C","",strsplit(filename[[1]][2],split=c("[_]"))[[1]][1])
    str<-paste(str,"NS_",num,sep="")
    str<-paste(str,"_1g",sep="")
    str_vec<-c(str_vec,str)
  }
}

check<-cbind(str_vec,colnames(allcounts_allcells))

#Write to a file with the correct names.
colnames(allcounts_allcells)<-str_vec
setwd("/Volumes/guacamole/Software/Final Scripts for Paper _ 01152015/Files")
write.table(allcounts_allcells,"All_spec_pops_counts_09222015_correctNames.txt")


#Removes neurosphere cells and writes to another table.
allcounts_allcells_noNS<-allcounts_allcells[,!grepl("NS_",colnames(allcounts_allcells))]
setwd("/Volumes/guacamole/Software/Final Scripts for Paper _ 01152015/Files")
write.table(allcounts_allcells_noNS,"All_spec_pops_counts_09222015_correctNames_noNS.txt")




#The following block of code takes only the cells which had ERCC spike ins included in the libraries
#and writes a table of counts for only those cells

setwd("/Volumes/guacamole/Software/Final Scripts for Paper _ 01152015/AllCounts/ERCC")
fileList=list.files(pattern=".counts.txt$")

#Combine the counts from each of the individual files to make one large counts matrix and save the counts matrix to
#a file.
allcounts_ERCC<-read.table(fileList[1],header=F)
colnames(allcounts_ERCC)[2]<-strsplit(fileList[1],split="[.]")[[1]][1]

for (i in 2:length(fileList)){
  temp = read.table(fileList[i],header=F)
  allcounts_ERCC=cbind(allcounts_ERCC,temp[,2])
  colnames(allcounts_ERCC)[i+1]<-strsplit(fileList[i],split="[.]")[[1]][1]
}

allcounts_ERCC<-allcounts_ERCC[-c((length(allcounts_ERCC[,1])-4):length(allcounts_ERCC[,1])),] #removes HTseq QC values from bottom of counts matrix
allcounts_ERCC_fin<-allcounts_ERCC[,2:length(allcounts_ERCC[1,])]
rownames(allcounts_ERCC_fin)<-allcounts_ERCC[,1]

str_vec<-vector()
for(i in 1:length(colnames(allcounts_ERCC_fin))){
  str<-""
  if(strsplit(colnames(allcounts_ERCC_fin)[i],split=c("[_]"))[[1]][1]=="PGE120"){
    filename<-strsplit(colnames(allcounts_ERCC_fin)[i],split=c("[_]"))
    num<-strsplit(filename[[1]][2],split=c("[-]"))[[1]][2]
    qual<-strsplit(filename[[1]][2],split=c("[-]"))[[1]][3]
    str<-paste(str,"aNSC_C_",num,"_",qual,sep="")
    str_vec<-c(str_vec,str)
  }else if(strsplit(colnames(allcounts_ERCC_fin)[i],split=c("[_]"))[[1]][1]=="PGE210"){
    filename<-strsplit(colnames(allcounts_ERCC_fin)[i],split=c("[_]"))
    num<-filename[[1]][3]
    qual<-filename[[1]][4]
    str<-paste(str,"aNSC_D_",num,"_",qual,sep="")
    str_vec<-c(str_vec,str)
  }else if(strsplit(colnames(allcounts_ERCC_fin)[i],split=c("[_]"))[[1]][1]=="PG210"){
    filename<-strsplit(colnames(allcounts_ERCC_fin)[i],split=c("[_]"))
    num<-filename[[1]][3]
    qual<-filename[[1]][4]
    str<-paste(str,"qNSC_C_",num,"_",qual,sep="")
    str_vec<-c(str_vec,str)
  }else if(strsplit(colnames(allcounts_ERCC_fin)[i],split=c("[-]"))[[1]][1]=="Na"){
    filename<-strsplit(colnames(allcounts_ERCC_fin)[i],split=c("[-]"))
    num<-gsub("C","",strsplit(filename[[1]][2],split=c("[_]"))[[1]][1])
    str<-paste(str,"aNSC_A_",num,sep="")
    str<-paste(str,"_1g",sep="")
    str_vec<-c(str_vec,str)
  }else if(strsplit(colnames(allcounts_ERCC_fin)[i],split=c("[-]"))[[1]][1]=="Nq"){
    filename<-strsplit(colnames(allcounts_ERCC_fin)[i],split=c("[-]"))
    num<-gsub("C","",strsplit(filename[[1]][2],split=c("[_]"))[[1]][1])
    str<-paste(str,"qNSC_A_",num,sep="")
    str<-paste(str,"_1g",sep="")
    str_vec<-c(str_vec,str)
  }else if(strsplit(colnames(allcounts_ERCC_fin)[i],split=c("[-]"))[[1]][1]=="PGE"){
    filename<-strsplit(colnames(allcounts_ERCC_fin)[i],split=c("[-]"))
    num<-gsub("C","",strsplit(filename[[1]][2],split=c("[_]"))[[1]][1])
    str<-paste(str,"aNSC_B_",num,sep="")
    str<-paste(str,"_1g",sep="")
    str_vec<-c(str_vec,str)
  }else if(strsplit(colnames(allcounts_ERCC_fin)[i],split=c("[-]"))[[1]][1]=="PG"){
    filename<-strsplit(colnames(allcounts_ERCC_fin)[i],split=c("[-]"))
    num<-gsub("C","",strsplit(filename[[1]][2],split=c("[_]"))[[1]][1])
    str<-paste(str,"qNSC_B_",num,sep="")
    str<-paste(str,"_1g",sep="")
    str_vec<-c(str_vec,str)
  }else if(strsplit(colnames(allcounts_ERCC_fin)[i],split=c("[-]"))[[1]][1]=="G"){
    filename<-strsplit(colnames(allcounts_ERCC_fin)[i],split=c("[-]"))
    num<-gsub("C","",strsplit(filename[[1]][2],split=c("[_]"))[[1]][1])
    str<-paste(str,"Ast_",num,sep="")
    str<-paste(str,"_1g",sep="")
    str_vec<-c(str_vec,str)
  }else if(strsplit(colnames(allcounts_ERCC_fin)[i],split=c("[-]"))[[1]][1]=="E"){
    filename<-strsplit(colnames(allcounts_ERCC_fin)[i],split=c("[-]"))
    num<-gsub("C","",strsplit(filename[[1]][2],split=c("[_]"))[[1]][1])
    str<-paste(str,"NPC_",num,sep="")
    str<-paste(str,"_1g",sep="")
    str_vec<-c(str_vec,str)
  }else if(strsplit(colnames(allcounts_ERCC_fin)[i],split=c("[-]"))[[1]][1]=="N"){
    filename<-strsplit(colnames(allcounts_ERCC_fin)[i],split=c("[-]"))
    num<-gsub("C","",strsplit(filename[[1]][2],split=c("[_]"))[[1]][1])
    str<-paste(str,"NS_",num,sep="")
    str<-paste(str,"_1g",sep="")
    str_vec<-c(str_vec,str)
  }
}

check<-cbind(str_vec,colnames(allcounts_ERCC_fin))


colnames(allcounts_ERCC_fin)<-str_vec
setwd("/Volumes/guacamole/Software/Final Scripts for Paper _ 01152015/Files")
write.table(allcounts_ERCC_fin,"ERCC_spec_pops_counts_09222015_correctNames_withSpikes.txt")

