#This script takes the counts matrices which were generated by the script 09232015_EstablishingCountsMatrix
#And performs quality filtering and QC plots which are included in Figure S1.

#The result of this script are final counts matrices which are filtered for high quality cells.

rm(list=ls())
library(ggplot2)
library(edgeR)


#----------------------------------------------------------------------------------------------------------------------------#
#----------------------------------------------------------------------------------------------------------------------------#

#All in vivo specific populations - genes detected

#Read in all counts and take only those cells with the 1g designation.
setwd("/Volumes/guacamole/Software/Final Scripts for Paper _ 01152015/Files")
allcounts<-read.table("All_spec_pops_counts_09222015_correctNames_noNS.txt",header=T)
allcounts_single<-allcounts[,grepl("_1g",colnames(allcounts))]

allcells_geneNum<-colSums(allcounts_single!= 0)   #Number of nonzero FPKM values for each cell
d=data.frame(Genes_Detected=allcells_geneNum[2:length(allcells_geneNum)])
p<-ggplot(d)
p<-p+geom_histogram(aes(x=d$Genes_Detected),binwidth=100, color="black",fill="red")
p<-p+geom_density(aes(x=d$Genes_Detected))
p<-p+labs(x="# of Genes Detected")
p<-p+theme_classic()
p<-p+theme(axis.text.x=element_text(size=18))
p<-p+theme(axis.title.x=element_text(size=22))
p<-p+theme(axis.text.y=element_text(size=18))
p<-p+theme(axis.title.y=element_text(size=22))
p<-p+theme(axis.title.y=element_text(vjust=1))
p<-p+theme(axis.title.x=element_text(vjust=-0.1))
p
#FIGURE S1B
setwd("/Volumes/guacamole/Software/Final Scripts for Paper _ 01152015/FigureS1_Output")
pdf("NumGenesDetect_allcells_miseq.pdf", width = 8, height=4)
print(p)
dev.off()

#All cells which have genes detected counts over 500
allcells_counts_genes_single<-allcounts_single[,allcells_geneNum>500]

#----------------------------------------------------------------------------------------------------------------------------#
#Number of reads mapped to genes

#Read in all counts and take only those cells with the 1g designation.
setwd("/Volumes/guacamole/Software/Final Scripts for Paper _ 01152015/Files")
allcounts<-read.table("All_spec_pops_counts_09222015_correctNames_noNS.txt",header=T)
allcounts_single<-allcounts[,grepl("_1g",colnames(allcounts))]

allcells_readsMap<-colSums(allcounts_single)   #Number of nonzero FPKM values for each cell
d=data.frame(Genes_Detected=allcells_readsMap[2:length(allcells_readsMap)])
p<-ggplot(d)
p<-p+geom_histogram(aes(x=d$Genes_Detected),binwidth=5000, color="black",fill="red")
p<-p+geom_density(aes(x=d$Genes_Detected))
p<-p+labs(x="# of Reads Mapped to Transcriptome")
p<-p+theme_classic()
p<-p+theme(axis.text.x=element_text(size=18))
p<-p+theme(axis.title.x=element_text(size=22))
p<-p+theme(axis.text.y=element_text(size=18))
p<-p+theme(axis.title.y=element_text(size=22))
p<-p+theme(axis.title.y=element_text(vjust=1))
p<-p+theme(axis.title.x=element_text(vjust=-0.1))
p
#FIGURE S1B
setwd("/Volumes/guacamole/Software/Final Scripts for Paper _ 01152015/FigureS1_Output")
pdf("NumReads_in_genes_Detect_allcells_miseq.pdf", width = 8, height=4)
print(p)
dev.off()


#All cells which have number of reads mapping to transcriptome over 20,000
allcells_counts_reads_single<-allcounts_single[,allcells_readsMap>20000]


#Determine the intersection of cells which have genes detected over 500 and reads mapping to transcriptome over 20,000.
intersect_names<-intersect(colnames(allcells_counts_reads_single),colnames(allcells_counts_genes_single))

#----------------------------------------------------------------------------------------------------------------------------#
#----------------------------------------------------------------------------------------------------------------------------#
#----------------------------------------------------------------------------------------------------------------------------#
#----------------------------------------------------------------------------------------------------------------------------#
#For the cells which have ERCC spike ins (Cells from 01202015 and 02102015) and filter cells that have less than 30% ERCC spike
#in contribution to their libraries.

#Read in all cells that have spike ins.
setwd("/Volumes/guacamole/Software/Final Scripts for Paper _ 01152015/Files")
allcounts<-read.table("ERCC_spec_pops_counts_09222015_correctNames_withSpikes.txt",header=T)
allcounts_single<-allcounts[,grepl("_1g",colnames(allcounts))]

#Count all reads along each of the columns
counts<-apply(allcounts_single[,1:length(allcounts_single[1,])],2,sum)

#Extract all ERCC spike ins and count them
ERCC<-allcounts_single[grep("ERCC-",rownames(allcounts_single)),]
counts_ERCC<-apply(ERCC[,1:length(ERCC[1,])],2,sum)

percent_ERCC<-counts_ERCC/counts


#Plot Histograms
g<-ggplot(data=data.frame(percent_ERCC))+geom_histogram(aes(x=percent_ERCC),stat="bin",binwidth=0.05,color='blue')
g<-g+labs(x="Fraction of Reads Occupied by ERCC RNA Spike-Ins")
g<-g+theme(axis.text=element_text(size=18))
g<-g+theme(axis.title=element_text(size=24))
g
setwd("/Volumes/guacamole/Software/Final Scripts for Paper _ 01152015/FigureS1_Output")
pdf("ERCC_percent_green.pdf",height=5,width=10)
print(g)
dev.off()


write.table(percent_ERCC,"Percent_ERCC_reads.txt")

#Filtering out cells that have percent ERCC > 30%
all_counts_ERCClowqual<-allcounts_single[,percent_ERCC>.30]

#Modify the list of cells that are high quality based on gene counts and reads detected based on the cells
#that also have acceptable levels of ERCC spike ins.
mod_intersect_names<-intersect_names[-na.omit(match(colnames(all_counts_ERCClowqual),intersect_names))]

match(colnames(all_counts_ERCClowqual),mod_intersect_names)


#ESTABLISHING FINAL COUNTS MATRIX
#----------------------------------------------------------------------------------------------------------------------------#
#----------------------------------------------------------------------------------------------------------------------------#
#----------------------------------------------------------------------------------------------------------------------------#
#----------------------------------------------------------------------------------------------------------------------------#
setwd("/Volumes/guacamole/Software/Final Scripts for Paper _ 01152015/Files")
allcounts<-read.table("All_spec_pops_counts_09222015_correctNames_noNS.txt",header=T)

#take all counts from cells that have been filtered for gene counts, mapping reads, and ERCC spike ins
allcounts_filtered<-allcounts[match(mod_intersect_names,colnames(allcounts))]

#This is the final counts matrix that is referenced in the majority of the subsequent files.
setwd("/Volumes/guacamole/Software/Final Scripts for Paper _ 01152015/Files")
write.table(allcounts_filtered,"AllCounts_specPops_read_gene_ERCC_filt_FINAL.txt")

